src/components/feedback/test/ToastMessages-test.js

```diff
@@ -0,0 +1,168 @@
+      message: 'Foobar',
+    },
+    {
+      id: '3',
+      type: 'error',
+      message: 'Something failed',
+    },
+  ];
+  let fakeOnMessageDismiss;
+
+  beforeEach(() => {
+    wrappers = [];
+    fakeOnMessageDismiss = sinon.stub();
+  });
+
+  afterEach(() => {
+    wrappers.forEach(wrapper => wrapper.unmount());
+  });
+
+  function createToastMessages(toastMessages, setTimeout) {
+    const container = document.createElement('div');
+    document.body.appendChild(container);
+
+    const wrapper = mount(
+      <ToastMessages
+        messages={toastMessages}
+        onMessageDismiss={fakeOnMessageDismiss}
+        setTimeout_={setTimeout}
+      />,
+      { attachTo: container },
+    );
+    wrappers.push(wrapper);
+
+    return wrapper;
+  }
+
+  function triggerAnimationEnd(wrapper, index, direction = 'out') {
+    wrapper
+      .find('BaseToastMessageTransition')
+      .at(index)
+      .props()
+      .onTransitionEnd(direction);
+    wrapper.update();
+  }
+
+  it('renders a list of toast messages', () => {
+    const wrapper = createToastMessages(toastMessages);
+    assert.equal(wrapper.find('ToastMessageItem').length, toastMessages.length);
+  });
+
+  toastMessages.forEach((message, index) => {
+    it('dismisses messages when clicked', () => {
+      const wrapper = createToastMessages(toastMessages);
+
+      wrapper.find('Callout').at(index).props().onClick();
+      // onMessageDismiss is not immediately called. Transition has to finish
+      assert.notCalled(fakeOnMessageDismiss);
+
+      // Once dismiss animation has finished, onMessageDismiss is called
+      triggerAnimationEnd(wrapper, index);
+      assert.calledWith(fakeOnMessageDismiss, message.id);
+    });
+  });
+
+  it('dismisses messages automatically unless instructed otherwise', () => {
+    const messages = [
+      ...toastMessages,
+      {
+        id: 'foo',
+        type: 'success',
+        message: 'Not to be dismissed',
+        autoDismiss: false,
+      },
+    ];
+    const wrapper = createToastMessages(
+      messages,
+      // Fake internal setTimeout, to immediately call its callback
+      callback => callback(),
+    );
+
+    // Trigger "in" animation for all messages, which will schedule dismiss for
+    // appropriate messages
+    messages.forEach((_, index) => {
+      triggerAnimationEnd(wrapper, index, 'in');
+    });
+
+    // Trigger "out" animation on components which "direction" prop is currently
+    // "out". That means they were scheduled for dismiss
+    wrapper
+      .find('BaseToastMessageTransition')
+      .forEach((transitionComponent, index) => {
+        if (transitionComponent.prop('direction') === 'out') {
+          triggerAnimationEnd(wrapper, index);
+        }
+      });
+
+    // Only one toast message will remain, as it was marked as `autoDismiss: false`
+    assert.equal(fakeOnMessageDismiss.callCount, 3);
+  });
+
+  it('schedules dismiss only once per message', async () => {
```

Comment:
This test is marked async, but contains no `await`. Is it missing something?

Reply:
Nop, I think it's just an overlook after refactoring the test. It works the same without `async`, so I've removed it.